### PR TITLE
Path complete given path literal

### PR DESCRIPTION
**Pull request by deeuu**
Hey,

Thanks for all your great work - this is my first PR to `xonsh`, so let's see how I do ;-)

This is a PR to enable path completion of path literals, which without can be very frustrating
```
path = p"some/super/duper/long/path/to/comple<TAB> # :-(
```

I've added some minor modifications such that the `p` prefix is removed from the string passed to `ast.literal_eval` in `xonsh.completers._path_from_partial_string`, and also removed from the start of the path used by `xonsh.completers.complete_path`.

Please note that I don't have a deep understanding of these functions, but have implemented a few tests.

Let me know what you think.

Many thanks

### CODE REVIEW

File: xonsh/completers/path.py [hunk 1]
```diff
@@ -280,7 +284,7 @@ def complete_path(prefix, line, start, end, ctx, cdpath=True, filtfunc=None):
     if p is not None:
         lprefix = len(p[0])
         prefix = p[1]
-        path_str_start = p[2]
+        path_str_start = p[2].lstrip("p")
```
Comment (scopatz) [inline xonsh/completers/path.py:287]: I think this might do the wrong thing for `fp""` strings

**Comment by scopatz** [comment]
Thanks @deeuu! This is an awesome feature to have!  I have the one question about how this interacts with p-strings when combined with other prefixes, like f-strings.  

Any other thoughts on this @gforsyth?

**Comment by gforsyth** [comment]
> Any other thoughts on this @gforsyth?

This looks good!  Thanks for putting this in @deeuu -- I have hit this particular issue myself so I'll enjoy having this in. As @scopatz mentioned, the only additional thing to cover here is that it is possible to have `pf"` or `fp"` (and the equivalent with single quotes).

I don't think that will complicate the prefix-stripping logic too much -- I think you'll want to check `if p in quote` and then adjust your strip index by the right number of characters.  Let me know if you hit any bumps!

File: xonsh/completers/path.py (hunk 1 above)
Comment (deeuu) [reply xonsh/completers/path.py:287]: Nothing changes for the case when the quote is `pf"` or `fp"`, as `_path_from_partial_string` returns `None` for these cases. More than happy to add a check before stripping though.

**Comment by deeuu** [comment]
Thanks both. Path completion will simply not work for `fp""` or `pf""` strings (see my response above).

Do we want to take things a step further and evaluate expresssions inside pf-strings, e.g.

```
a = "very"
b = "nice"
path = pf"~/{a}/{b}<TAB>
```
Happy to give this a stab if you wouldn't mind show me how to parse/evaluate pf-strings in xonsh!

**Comment by scopatz** [comment]
> Path completion will simply not work for

Yeah, agreed. we don't want to start doing evaluation like that during completion because it could have too many side effects.  I think the concern here is just that `pf` strings were not being skipped properly in the logic.  Hope that helps clear it up!

**Comment by laloch** [comment]
Hi folks, what's the status? I'd like the feature to get merged.

**Comment by deeuu** [comment]
Hey, 

Hoping to finish this tomorrow, completion tests are currently failing on Windows so I need to investigate (I'm not on Windows but have created a VM for xonsh dev  :smile: )

**Comment by deeuu** [comment]
I've made some improvements and path completion now works with path strings (`p""`) and raw path strings (`pr""`).

Unfortunately, the path completion test for this [is failing on Windows](https://github.com/xonsh/xonsh/blob/685ab1ba30b73e6bf2056b49a74f89adf8e0bb57/tests/test_path_completers.py#L55), but I can't reproduce on Windows 10 64-bit. Any ideas?

I suspect it's to an issue with backslashes in the temp-file name, but thought I'd gotten around that by [using a raw path string in the prefix](https://github.com/xonsh/xonsh/blob/685ab1ba30b73e6bf2056b49a74f89adf8e0bb57/tests/test_path_completers.py#L55) (at least that worked on my Windows VM!).

**Comment by scopatz** [comment]
Very cool! Thanks for plugging away at this @dyuri!

**Comment by scopatz** [comment]
FWIW, these windows errors seem like legitimate errors

**Comment by deeuu** [comment]
After further digging by abusing the Windows CI, it looks like [`_iglobpath`](https://github.com/xonsh/xonsh/blob/master/xonsh/tools.py#L2460) is returning an empty set after globbing a case expanded path such as `C:\Users\RUNNER~1\AppData\Local\Temp\tmpcuo54ym8*` -> `C:\[Uu][Ss]...`.

@scopatz No need for a review (sorry for the notificatiion), but any ideas?

**Comment by deeuu** [comment]
Okay, so Windows file system is case-insensitive (see [this discussion](https://bugs.python.org/issue26655) on `pathlib` `glob`), so running the test with `$CASE_SENSITIVE_COMPLETIONS=True` resolves the issue.

**Comment by gforsyth** [comment]
Thanks for sticking with this through some thorny testing, @deeuu !

**Comment by gforsyth** [comment]
And congrats on your first PR, and welcome!

**Pull request merged**